src/approaches/interactive_learning_approach.py

```diff
@@ -232,6 +239,25 @@ def _termination_function(s: State) -> bool:
 
         return act_policy, _termination_function
 
+    def _create_random_interaction_strategy(
+        self, train_task_idx: int
+    ) -> Tuple[Callable[[State], Action], Callable[[State], bool]]:
+        """Sample and execute random initiable options until timeout."""
+
+        random_options_approach = RandomOptionsApproach(
```

Comment:
initialize this in init? we don't need a new object each time

Reply:
oh, the get_current_predicates() might change though

Reply:
right